mmdet/datasets/custom.py

```diff
@@ -72,24 +73,31 @@ def __init__(self,
         self.test_mode = test_mode
         self.filter_empty_gt = filter_empty_gt
         self.CLASSES = self.get_classes(classes)
+        self.file_client = mmcv.FileClient(**file_client_args)
 
         # join paths if data_root is specified
         if self.data_root is not None:
             if not osp.isabs(self.ann_file):
-                self.ann_file = osp.join(self.data_root, self.ann_file)
+                self.ann_file = self.file_client.join_path(
```

Comment:
No need to modify this part of the code, because `get_local_path` will complete this logic